src/operator/nn/softmax-inl.h

```diff
@@ -127,7 +137,7 @@ inline void SoftmaxGrad(Stream<cpu> *s, DType *out, DType *ograd,
 #ifdef __CUDACC__
 template<int x_bits, typename OP, typename DType, int ndim>
 __global__ void softmax_compute_kernel(DType *in, DType *out, index_t M, int axis,
-                                       Shape<ndim> sshape, Shape<ndim> stride) {
+                                       Shape<ndim> sshape, Shape<ndim> stride, float temperature) {
```

Comment:
Please try to use const for variables which don't change

Reply:
This is pass-by-value so it does not make any difference.

Reply:
Just a matter of style https://google.github.io/styleguide/cppguide.html#Use_of_const

Reply:
Thanks for your suggestion. But I do not seem to find the convention of using const for pass-by-value parameter in the Google style guide: 
```
If a function guarantees that it will not modify an argument passed by reference or by pointer, the corresponding function parameter should be a reference-to-const (const T&) or pointer-to-const (const T*), respectively.
```
In fact, adding unnecessary const declaration will put restriction on the caller of the library function.

Reply:
adding `const` qualifier could be a safe net by explicitly telling the compiler about your assumption that you're not changing the input value at all in the function.
Say if you have a function like this:
```c++
void foo(int a) {
  return a;
}
```
Here your assumption that the original input should be returned without any changes, and that is essential for correct behavior of this function.
Now if someone happens to change the function to:
```c++
void foo(int a) {
  a++;      // <- new code that happens to change the value of a, and will affect correctness
  return a;
}
```
compiler will not complain about it as you do not have a `const` qualifier here, so an extra `const` qualifier is not necessary, but it could be helpful.

Reply:
I have added const qualifier following your suggestion.